platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java

```diff
@@ -0,0 +1,180 @@
+
+import static com.swirlds.logging.legacy.LogMarker.RECONNECT;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.stream.RunningEventHashOverride;
+import com.swirlds.logging.legacy.LogMarker;
+import com.swirlds.platform.components.AppNotifier;
+import com.swirlds.platform.components.SavedStateController;
+import com.swirlds.platform.consensus.EventWindow;
+import com.swirlds.platform.event.AncientMode;
+import com.swirlds.platform.event.validation.AddressBookUpdate;
+import com.swirlds.platform.eventhandling.EventConfig;
+import com.swirlds.platform.listeners.ReconnectCompleteNotification;
+import com.swirlds.platform.state.SwirldStateManager;
+import com.swirlds.platform.state.nexus.SignedStateNexus;
+import com.swirlds.platform.state.signed.SignedState;
+import com.swirlds.platform.system.InitTrigger;
+import com.swirlds.platform.system.Platform;
+import com.swirlds.platform.system.address.AddressBook;
+import com.swirlds.platform.system.address.AddressBookUtils;
+import com.swirlds.platform.system.status.actions.ReconnectCompleteAction;
+import com.swirlds.platform.wiring.PlatformWiring;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+// FUTURE WORK: this data should be traveling out over the wiring framework.
+
+/**
+ * Encapsulates the logic for loading the reconnect state.
+ */
+public class ReconnectStateLoader {
+
+    private static final Logger logger = LogManager.getLogger(ReconnectStateLoader.class);
+    private final Platform platform;
+    private final PlatformContext platformContext;
+    private final PlatformWiring platformWiring;
+    private final SwirldStateManager swirldStateManager;
+    private final SignedStateNexus latestImmutableStateNexus;
+    private final SavedStateController savedStateController;
+    private final AddressBook addressBook;
+
+    /**
+     * Constructor.
+     *
+     * @param platform                  the platform
+     * @param platformContext           the platform context
+     * @param platformWiring            the platform wiring
+     * @param swirldStateManager        manages the mutable state
+     * @param latestImmutableStateNexus holds the latest immutable state
+     * @param savedStateController      manages how states are saved
+     * @param addressBook               the address book
+     */
+    public ReconnectStateLoader(
+            @NonNull final Platform platform,
+            @NonNull final PlatformContext platformContext,
+            @NonNull final PlatformWiring platformWiring,
+            @NonNull final SwirldStateManager swirldStateManager,
+            @NonNull final SignedStateNexus latestImmutableStateNexus,
+            @NonNull final SavedStateController savedStateController,
+            @NonNull final AddressBook addressBook) {
+        this.platform = platform;
```

Comment:
missing `Objects.requireNonNull` for all these args

Reply:
added